### PR TITLE
Add thermo capillary two droplet simulation

### PR DESCRIPTION
**Pull request by nmuch**
This PR adds the two-droplet simulation case from [Nas, Tryggvason 2003](https://www.sciencedirect.com/science/article/pii/S0301932203000843?casa_token=Sw9xFN3LLFIAAAAA:6U4NSOa_hNtNYYvYM8d4ctMo_M6XfzqLOYRFTfctMUa5cmOLFKXPisJpgGqvB2fu66AnxL3yWyE). 
Also, the level-set field is now passed to the heat operation.

Here, the result is plotted for t_r = 3.8. The lines are iso-thermals.
image

Some issue still remain:
- The side boundary conditions should be periodic. I'll add periodic boundary conditions in a follow-up PR.
- The top/bottom boundary conditions don't seem to correlate to what's described in the paper.
- The (heat and fluid) material parameters are distributed across multiple sections in the Parameters struct. I'll unify the definition of material parameters for multiple physics using the material struct from #107 in a follow-up PR.
- The simulation result does not correlate to the reference solution after some time. The image below shows the result for  t_r = 25.3. This problem may be resolved by the correct boundary conditions.
image

### CODE REVIEW

File: tests/thermo_capillary_droplet.with_adaflo_support=on.mpirun=4.output [hunk 1]
```diff
@@ -1,80 +1,88 @@
-| normal vector:         i=11 	|n_0| = 60206.369632   |n_1| = 60206.369632
```
Comment (nmuch) [inline tests/thermo_capillary_droplet.with_adaflo_support=on.mpirun=4.output:1]: This test's result changed, because the heat transfer now uses the level-set field, too. fyi @mschreter

File: include/meltpooldg/simulations/thermo_capillary_two_droplets/thermo_capillary_two_droplets.hpp [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      this->attach_no_slip_boundary_condition(lower_bc, "navier_stokes_u");
+      this->attach_no_slip_boundary_condition(upper_bc, "navier_stokes_u");
+      this->attach_symmetry_boundary_condition(left_bc, "navier_stokes_u");
+      this->attach_symmetry_boundary_condition(right_bc, "navier_stokes_u");
+
+      this->attach_dirichlet_boundary_condition(lower_bc,
+                                                std::make_shared<Functions::ConstantFunction<dim>>(
+                                                  reference_temperautre),
+                                                "heat_transfer");
+      this->attach_dirichlet_boundary_condition(upper_bc,
```
Comment (mschreter) [inline include/meltpooldg/simulations/thermo_capillary_two_droplets/thermo_capillary_two_droplets.hpp:220]: So far, dirichlet boundary conditions were not set in the `two_phase_flow_with_heat_transfer_problem`. I've fixed the bug in PR #112. Maybe you can copy the relevant lines in this PR, then we can first merge this one.

File: include/meltpooldg/simulations/thermo_capillary_two_droplets/thermo_capillary_two_droplets.hpp [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+ *    cp     = 1e-4 J/kg/K
+ *
+ * surface tension coefficient: 0.025239459 N/m
+ * temperature-dependent surface tension coefficient: 0.005047892 N/m/K
+ * ∇T = 4.754459964 K/m
+ * T1 = 290 K
+ * T2 = 290 + 16 * a * ∇T = 293.333 K
+ *
+ * reference scales
+ *    Ur = 0.043817804
```
[the remark this answers is not in the document]
Comment (nmuch) [reply include/meltpooldg/simulations/thermo_capillary_two_droplets/thermo_capillary_two_droplets.hpp:72]: Sure.
Ur is the reference velocity 
Also, tr is the reference time scale 
I'll add this to the comment.
